examples/step-3/doc/results.dox

```diff
@@ -224,3 +224,213 @@ suggestions:
+     data_file.write_dataset("point_value", point_value);
+     Vector<double> mean_value(1);
+     mean_value[0] = VectorTools::compute_mean_value(dof_handler, QGauss<2>(fe.degree + 1), solution, 0);
+     data_file.write_dataset("mean_value",mean_value);
+     @endcode
+     
+<h5> Using R and ggplot2 to generate PDFs with plots of the grid, the solution and convergence curves: </h5>
+     If you are unfamiliar with R and ggplot2 you could check out the data carpentry course on R
+     <a href=https://datacarpentry.org/R-ecology-lesson/index.html>here</a>. <br>
+     Furthermore, since most search engines struggle with searches of the form "R + topic",
+     we recommend using the specialized service <a href=https://www.rseek.org> RSeek </a> instead.
+     The most prominent difference between R and other languages is that the assignment operator (a = 5) is interchangable with
+     (a <- 5). As the latter is considered standard we will use it in our examples as well.
+     To open the .h5 file in R you have to install the <a href=https://bioconductor.org/packages/release/bioc/html/rhdf5.html>rhdf5</a> package, which is a part of the Bioconductor package.
+     <br>First we will include all necessary packages and have a look at how the data is structured in our file.
+     @code
+     library(rhdf5) #library for handling HDF5 files
+     library(ggplot2) #main plotting library
+     library(grDevices) #needed for output to PDF
+     library(viridis) #contains good colormaps for sequential data
+     refinement <- 5
+     h5f <- H5Fopen(paste("solution_",refinement,".h5",sep=""))
+     print(h5f)
+     @endcode
+     This gives the following output
+     @code
+     HDF5 FILE 
+        name /
+    filename 
+
+         name       otype  dclass     dim
+     0 cells       H5I_DATASET INTEGER  x 1024
+     1 mean_value  H5I_DATASET FLOAT   1      
+     2 nodes       H5I_DATASET FLOAT    x 1089
+     3 point_value H5I_DATASET FLOAT   1      
+     4 solution    H5I_DATASET FLOAT    x 1089
+     @endcode
+     The datasets can be accessed by <code>h5f\$name</code>. The function <code>dim(h5f\$cells)</code> gives us
+     the dimensions of our cell matrix. We can see the following three matrices, as well as the two additional
+     data points we added.
+     <ul> 
+     <li> <code>cells</code> a 4x1024 matrix that stores the  (C++) vertex indices for each cell 
+     <li> <code>nodes</code> a 2x1089 matrix storing the position values (x,y) for our cell vertices
+     <li> <code>solution</code> a 1x1089 matrix storing the values of our solution at each vertex
+     </ul>
+     Now we can use this data to generate our various plots. Plotting with ggplot2 usually splits into two steps.
+     At first the data needs to be manipulated and added to a <code>data.frame</code>.
+     After that a <code>ggplot</code> object is constructed and manipulated by adding plot elements to it.
+     <ul>
+     <li>
```

Comment:
This works but it is not the style we use when we list out example code. I would just remove the section names and list the paragraphs in sequence without bullet points.